src/rgw/rgw_rest_s3.cc

```diff
@@ -6338,47 +6509,109 @@ int RGWSelectObj_ObjStore_S3::run_s3select(const char* query, const char* input,
       csv.escape_char = *m_escape_char.c_str();
     }
 
+    if (m_enable_progress.compare("true")==0) {
+      enable_progress = true;
+    } else {
+      enable_progress = false;
+    }
+
+    if (output_row_delimiter.size()) {
+      csv.output_row_delimiter = *output_row_delimiter.c_str();
+    }
+
+    if (output_column_delimiter.size()) {
+      csv.output_column_delimiter = *output_column_delimiter.c_str();
+    }
+
+    if (output_quot.size()) {
+      csv.output_quot_char = *output_quot.c_str();
+    }
+
+    if (output_escape_char.size()) {
+      csv.output_escape_char = *output_escape_char.c_str();
+    }
+
+    if(output_quote_fields.compare("ALWAYS") == 0) {
+      csv.quote_fields_always = true;
+    }
+    else if(output_quote_fields.compare("ASNEEDED") == 0) {
+      csv.quote_fields_asneeded = true;
+    }
+
     if(m_header_info.compare("IGNORE")==0) {
       csv.ignore_header_info=true;
     }
     else if(m_header_info.compare("USE")==0) {
       csv.use_header_info=true;
     }
-
     m_s3_csv_object = std::unique_ptr<s3selectEngine::csv_object>(new s3selectEngine::csv_object(s3select_syntax.get(), csv));
   }
 
-  header_size = create_header_records(m_buff_header.get());
-  m_result.append(m_buff_header.get(), header_size);
-  m_result.append(PAYLOAD_LINE);
+  m_aws_response_handler.get()->init_response();
```

Comment:
you can use -> directly on unique_ptrs: `m_aws_response_handler->init_response()`

same below